go/rpcclientlib/viewing_key_manager.go

```diff
@@ -0,0 +1,32 @@
+package rpcclientlib
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto/ecies"
+)
+
+// ViewingKeyManager interface allows us to provide different versions of the viewing key client. The initial motivation for this was:
+// - we want the go-obscuro AuthClient to always be bound to a single account for code that it easy to reason about
+// - we want the WalletExtension to
+type ViewingKeyManager interface {
```

Comment:
I feel the responsibilities are not there yet.

I'd suggest getting rid of this interface.

The SingleAccountVK.. - should become the "ViewingKey" - containing all relevant data, including the decrypt function.

Then, we create the "MultiAccountManager" - or something like that which is pretty much the MultiViewingKey.., responsible with helping use the right VK